include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -561,7 +561,8 @@ make sure that environment variable OPENPMD_BP_BACKEND is not set to ADIOS1.
             private:
                 using UnitType = typename FieldTmp::UnitValueType;
                 using ValueType = typename FieldTmp::ValueType;
-                using ComponentType = typename GetComponentsType<ValueType>::type;
+                using GetComponentsTypeValueType = GetComponentsType<ValueType>;
+                using ComponentType = typename GetComponentsTypeValueType::type;
```

Comment:
Consider adding a comment that this is a workaround for a bug here. Otherwise, if I would see this code, I would refactor it back :)

Reply:
Yup, I noticed that too after submitting. Will add a comment if we go with this change.